components/action-mgt/org.wso2.carbon.identity.action.management/src/main/java/org/wso2/carbon/identity/action/management/api/model/BinaryObject.java

```diff
@@ -18,32 +18,75 @@
 
 package org.wso2.carbon.identity.action.management.api.model;
 
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
 import org.wso2.carbon.identity.action.management.api.exception.ActionMgtRuntimeException;
 
+import java.io.BufferedReader;
+import java.io.ByteArrayInputStream;
 import java.io.IOException;
 import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
 
 /**
  * This class models the object typed action property values
  */
 public class BinaryObject {
 
-    private final InputStream value;
+    private String stringValue;
+    private InputStream streamValue;
 
-    public BinaryObject(InputStream value) {
+    private BinaryObject(String value) {
 
-        this.value = value;
+        this.stringValue = value;
     }
 
-    public InputStream getInputStream() {
+    private BinaryObject(InputStream value) {
 
-        return value;
+        this.streamValue = value;
+    }
+
+    public static BinaryObject convertInputStreamToString(InputStream inputStream) {
```

Comment:
Even this appeared as static methods, these are real constructor calls for the objects. I prefer call these methods via constructors. Why do we need to keep constructors private ?

Reply:
When someone tries to create "BinaryObject" objects, its not intuitive how to create the objects.